### PR TITLE
Allow not to configure semian for Mysql2

### PR DESCRIPTION
**Pull request by byroot**
I initially wrote that because I though we needed it, but actually we dont.

But maybe it still make sense.

@Sirupsen thoughts?

### CODE REVIEW

**Comment by Sirupsen** [comment]
I don't think we should encourage this behaviour if we can avoid it.

File: lib/semian/mysql2.rb [hunk 1]
```diff
@@ -62,7 +62,15 @@ def connect(*args)
     end
 
     def semian_resource
-      @semian_resource ||= ::Semian.retrieve_or_register(semian_identifier, **semian_options)
+      @semian_resource ||= case query_options[:semian]
+      when false
+        UnprotectedResource.new(semian_identifier)
```
Comment (Sirupsen) [inline lib/semian/mysql2.rb]: What about for Redis etc?

Comment (Sirupsen) [inline lib/semian/mysql2.rb]: Also do you have any ideas on how we don't have to implement this in every driver? That kind of sucks. I don't have any off the bat though.

**Comment by byroot** [comment]
@Sirupsen and @fw42 I rebased that PR if you guys want to take a look.

File: lib/semian/unprotected_resource.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+module Semian
+  class UnprotectedResource
```
Comment (Sirupsen) [inline lib/semian/unprotected_resource.rb]: Consider a short comment describing the intention of this class.

File: lib/semian/unprotected_resource.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    attr_reader :name
+
+    def initialize(name)
+      @name = name
+    end
+
+    def tickets
+      0
```
Comment (Sirupsen) [inline lib/semian/unprotected_resource.rb]: `-1` perhaps? Just to distinguish completely cleanly in monitoring. `0` is possible in the semaphore, `-1` is completely impossible.

**Comment by Sirupsen** [comment]
Two minor, optional, comments. LGTM!

**Comment by Sirupsen** [comment]
Re my previous comment: @byroot convinced me we need this.

**Pull request merged**